### PR TITLE
Patch ESLZ ARM

### PR DESCRIPTION
**Pull request by krowlandson**
## Overview/Summary

Minor updates to the ESLZ ARM templates to fix a few issues observed during development of the Terraform Module.

## This PR fixes/adds/changes/removes

1. Fix naming for lzsManagementGroup variable
1. Fix value for sqlEncryptionPolicyAssignment variable
1. Update name for the following policies to improve clarity on purpose:
    1. `Deny-Priv-Esc-AKS` --> `Deny-Priv-Escalation-AKS`
    1. `Deny-Privileged-AKS` --> `Deny-Priv-Containers-AKS`
1. Update `description` for `Deny-Subnet-Without-Nsg` Policy Assignment
1. Update `metadata.displayName` for `CognitiveServicesCMKEffect` parameter in `Enforce-Encryption-CMK` Policy Set Definition to remove unusual whitespace character (discovered during testing, as per the below image)

image

### Breaking Changes

n/a

## Testing Evidence

These changes replicate values set in a working deployment using the Terraform version of ES.

For the update on `Enforce-Encryption-CMK`, please see the below evidence of how this change effects the resource in question:

image

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Enterprise-Scale/pulls)
- [x] Associated it with relevant [issues](https://github.com/Azure/Enterprise-Scale/issues), for tracking and closure.
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Enterprise-Scale/tree/main)
- [x] Performed testing and provided evidence.
- [x] Updated relevant and associated documentation.
- [x] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located: `/docs/wiki/whats-new.md`)

### CODE REVIEW

**Comment by krowlandson** [comment]
I don't believe there are any specific updates needed on the "What's new" page as these are all covered under the existing descriptions, but please let me know if you think it needs an addition.

**Review by jtracey93 — DISMISSED** [review]
All looks good to me @krowlandson!

Just a couple of questions/notes:

1. Did you also change the Azure Backup Policy Assignment variable as the `azBackupIdentityPolicyAssignment` variable was never used?
2. You dont need to update the RIs in the `docs/reference/xxxx/` folders now for Adventure Works, Contoso, WingTip & Trey separately anymore; only the files in `eslzArm/` 👍

File: docs/reference/adventureworks/armTemplates/auxiliary/policies.json [hunk 1]
```diff
@@ -20173,7 +20173,7 @@
                   "Disabled"
                 ],
                 "metadata": {
-                  "displayName": "Cognitive Services accounts should enable data encryption with a customer-managed key (CMK)",
+                  "displayName": "Cognitive Services accounts should enable data encryption with a customer-managed key (CMK)",
                   "description": "Customer-managed keys (CMK) are commonly required to meet regulatory compliance standards. CMKs enable the data stored in Cognitive Services to be encrypted with an Azure Key Vault key created and owned by you. You have full control and responsibility for the key lifecycle, including rotation and management. Learn more about CMK encryption at https://aka.ms/cosmosdb-cmk."
                 }
```
Comment (krnese) [inline docs/reference/adventureworks/armTemplates/auxiliary/policies.json:20178]: We are no longer using this template.

File: docs/reference/contoso/armTemplates/auxiliary/policies.json [hunk 2]
```diff
@@ -20173,7 +20173,7 @@
                   "Disabled"
                 ],
                 "metadata": {
-                  "displayName": "Cognitive Services accounts should enable data encryption with a customer-managed key (CMK)",
+                  "displayName": "Cognitive Services accounts should enable data encryption with a customer-managed key (CMK)",
                   "description": "Customer-managed keys (CMK) are commonly required to meet regulatory compliance standards. CMKs enable the data stored in Cognitive Services to be encrypted with an Azure Key Vault key created and owned by you. You have full control and responsibility for the key lifecycle, including rotation and management. Learn more about CMK encryption at https://aka.ms/cosmosdb-cmk."
                 }
```
Comment (krnese) [inline docs/reference/contoso/armTemplates/auxiliary/policies.json:20178]: We are no longer using this template.

File: docs/reference/treyresearch/armTemplates/auxiliary/policies.json [hunk 3]
```diff
@@ -20173,7 +20173,7 @@
                   "Disabled"
                 ],
                 "metadata": {
-                  "displayName": "Cognitive Services accounts should enable data encryption with a customer-managed key (CMK)",
+                  "displayName": "Cognitive Services accounts should enable data encryption with a customer-managed key (CMK)",
                   "description": "Customer-managed keys (CMK) are commonly required to meet regulatory compliance standards. CMKs enable the data stored in Cognitive Services to be encrypted with an Azure Key Vault key created and owned by you. You have full control and responsibility for the key lifecycle, including rotation and management. Learn more about CMK encryption at https://aka.ms/cosmosdb-cmk."
                 }
               },
```
Comment (krnese) [inline docs/reference/treyresearch/armTemplates/auxiliary/policies.json:20179]: We are no longer using this template.

File: eslzArm/eslzArm.json [hunk 4]
```diff
@@ -607,8 +607,7 @@
...
             "azVmMonitorPolicyAssignment": "[uri(deployment().properties.templateLink.uri, 'managementGroupTemplates/policyAssignments/DINE-VMMonitoringPolicyAssignment.json')]",
             "azVmssMonitorPolicyAssignment": "[uri(deployment().properties.templateLink.uri, 'managementGroupTemplates/policyAssignments/DINE-VMSSMonitoringPolicyAssignment.json')]",
-            "azBackupLzPolicyAssignment": "[uri(deployment().properties.templateLink.uri, 'managementGroupTemplates/policyAssignments/DINE-VMBackupPolicyAssignment.json')]",
-            "azBackupIdentityPolicyAssignment": "[uri(deployment().properties.templateLink.uri, 'managementGroupTemplates/policyAssignments/DINE-VMBackupPolicyAssignment.json')]",
+            "azVmBackupPolicyAssignment": "[uri(deployment().properties.templateLink.uri, 'managementGroupTemplates/policyAssignments/DINE-VMBackupPolicyAssignment.json')]",
             "azPolicyForAksPolicyAssignment": "[uri(deployment().properties.templateLink.uri, 'managementGroupTemplates/policyAssignments/DINE-AksPolicyPolicyAssignment.json')]",
```
Comment (krnese) [inline eslzArm/eslzArm.json:611]: If identity subscription is enabled, we also recommend to assign VM Backup policy to the identity management group, hence we need a deterministic guid for the subsequent role assignment for that policy, so we need both these variables.

File: docs/reference/contoso/armTemplates/auxiliary/policies.json (hunk 2 above)
Comment (krowlandson) [reply docs/reference/contoso/armTemplates/auxiliary/policies.json:20178]: Any harm in this fix?

File: docs/reference/adventureworks/armTemplates/auxiliary/policies.json (hunk 1 above)
Comment (krowlandson) [reply docs/reference/adventureworks/armTemplates/auxiliary/policies.json:20178]: Any harm in this fix?

File: docs/reference/treyresearch/armTemplates/auxiliary/policies.json (hunk 3 above)
Comment (krowlandson) [reply docs/reference/treyresearch/armTemplates/auxiliary/policies.json:20179]: Any harm in this fix?

File: eslzArm/eslzArm.json (hunk 4 above)
Comment (krowlandson) [reply eslzArm/eslzArm.json:611]: As `azBackupIdentityPolicyAssignment` isn't referenced anywhere else in the repository and the values are the same, how does this make a difference when an identity subscription is enabled?

File: docs/reference/treyresearch/armTemplates/auxiliary/policies.json (hunk 3 above)
Comment (krnese) [reply docs/reference/treyresearch/armTemplates/auxiliary/policies.json:20179]: No harm, but we are planning to delete these files so I don't see any reason to why we need to update them

File: eslzArm/eslzArm.json (hunk 4 above)
Comment (krnese) [reply eslzArm/eslzArm.json:611]: It is it's own (optional) deployment name, and we cannot have duplicated resource name in the same ARM template.

Comment (jtracey93) [reply eslzArm/eslzArm.json:611]: Just my 2 cents here 😃 

I have tested a deployment from @krowlandson's branch for this PR and it works 👍 
image

Including the Backup policies involved also:
image

From my initial review, and re-review today, the ARM deployment names are unchanged just the reference to the deployment template URI and a de-duplication of a variable that holds this (which from looking at the current templates, it looks like the `azBackupIdentityPolicyAssignment` variable is not referenced anywhere as the `azBackupLzPolicyAssignment` is used for all nested deployments which do the assignments. Also as the conditionals are not based on these and other parameters/variables from the portal UX experience then I think this is all good 👍

**Review by krnese — APPROVED** [review]

**Pull request merged**